.github/workflows/publish-charts.yaml

```diff
@@ -0,0 +1,88 @@
+name: publish-charts
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Repository from where we want to build Charts'
+        required: true
+      repo_ref:
+        description: 'The branch/tag that will be used as new version for charts. Usually tag ref'
+        required: true
+      repo_chart:
+        description: 'Object with source path, destination path and chart name'
+        required: true
+      currentChartVersion:
+        description: 'Chart version set in `Chart.yaml`. Default: `LATEST`'
+        required: false
+        default: 'LATEST'
+
+env:
+  REPO: ${{ github.event.inputs.repo }}
+  REPO_REF: ${{ github.event.inputs.repo_ref }}
+  CurrentChartVersion: ${{ github.event.inputs.currentChartVersion }}
+
+jobs:
+  publish-charts:
+    runs-on: ubuntu-latest
+    name: Publish charts
+    continue-on-error: true
+    strategy:
+      max-parallel: 1
+      matrix: ${{ fromJson(github.event.inputs.repo_chart) }}
+    steps:
+
+      - id: checkout-gh-pages
+        uses: actions/checkout@v2 # Checkout Charts - gh-pages
+        with:
+          ref: 'gh-pages'
+          fetch-depth: 0
+
+      - id: checkout-helm-src
+        uses: actions/checkout@v2 # Checkout Helm source code
+        with:
+          repository: ${{ env.REPO }}
+          ref: ${{ env.REPO_REF }}
+          path: ${{ env.REPO }}
+
+      - id: chart-version
+        name: 'Set version - ${{ env.REPO_REF }}'
+        if: ${{ steps.checkout-gh-pages.outcome == 'success' && steps.checkout-helm-src.outcome == 'success' }}
+        run: |
+          find . \( -name "Chart.yaml" -o -name "values.yaml" \) -exec sed -i s/${{ env.CurrentChartVersion }}/${{ env.REPO_REF }}/ {} + 
```

Comment:
I've seen on another PR a comment regarding `yq`. Let's see if we can apply it here as well

Reply:
I've tried a few options locally but I can't make it work with one command (like the current one) to search and execute replace values for multiple files (different locations)
I think it would be too complex to have more than 1 command to replace this current one
p.s I've checked and `yq` is already installed on GH runners 🤞